### PR TITLE
Feature/xml export

### PR DESCRIPTION
**Pull request by modass**
Add an xml-exporter which allows to export to Uppaal-xml format

* based on tinyxml2
* single, selected components can be transformed individually
* a composition of ta's can be composed to a complete uppaal-system

### CODE REVIEW

File: test/test_xml_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	CHECK(res.find("</label>") != std::string::npos);
+	CHECK(res.find("x &lt; 3") != std::string::npos);
+}
+
+TEST_CASE("Write transition to xml", "[io]")
+{
+	tinyxml2::XMLDocument doc;
+	auto *                root_element = doc.NewElement("root");
+	doc.InsertEndChild(root_element);
+	automata::ta::Transition<std::string, std::string> transition{ automata::ta::Location<std::string>("l0"), "a", automata::ta::Location<std::string>("l1"), {}, {}};
```
Comment (morxa) [inline test/test_xml_writer.cpp:57]: That's a long line!

**Review by morxa — DISMISSED** [review]
Some minor nitpicking, but looks good otherwise :)

File: src/utilities/include/io/XmlWriter.h [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+		   << std::get<automata::AtomicClockConstraintT<std::less<automata::Time>>>(guard.second)
+		        .get_comparand();
+	} else if (std::holds_alternative<automata::AtomicClockConstraintT<std::greater<automata::Time>>>(
+	             guard.second)) {
+		ss << " > "
+		   << std::get<automata::AtomicClockConstraintT<std::less<automata::Time>>>(guard.second)
+		        .get_comparand();
+	}
+	auto tmp = ss.str();
+	xml_guard->SetText(tmp.c_str());
```
[the remark this answers is not in the document]
Comment (modass) [reply src/utilities/include/io/XmlWriter.h:59]: Jup - there was a thing about ss.str().c_str(), but I obviously did not read the article to the end. This is fine.

File: src/utilities/include/io/XmlWriter.h [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+	             guard.second)) {
+		ss << " >= "
+		   << std::get<automata::AtomicClockConstraintT<std::less<automata::Time>>>(guard.second)
+		        .get_comparand();
+	} else if (std::holds_alternative<automata::AtomicClockConstraintT<std::greater<automata::Time>>>(
+	             guard.second)) {
+		ss << " > "
+		   << std::get<automata::AtomicClockConstraintT<std::less<automata::Time>>>(guard.second)
+		        .get_comparand();
+	}
```
[the remark this answers is not in the document]
Comment (modass) [reply src/utilities/include/io/XmlWriter.h:57]: Valid, thanks

[the remark this answers is not in the document]
Comment (modass) [reply src/utilities/include/io/XmlWriter.h:57]: Could use a visitor, though.

**Review by morxa — APPROVED** [review]
Looks good, thank you :)

**Pull request merged**